### PR TITLE
[SuperLU_DIST]: v8.1.2

### PR DESCRIPTION
**Pull request by boriskaus**
bumps the version

### CODE REVIEW

**Comment by eschnett** [comment]
I do not know. I see that there is a linker failure. `libgfortran.so.5` is required, but is not found. It is possible that this library is required for MPItrampoline, and SuperLU_DIST picks up the requirement, but does not properly propagate the respective library directory from MPItrampoline's pkgconfig script (or however SuperLU_DIST queries the MPI implementation about its dependencies).

Actually, the error is reported by BinaryBuilder who can't track down the library, not by SuperLU_DIST's build script.

It might be that adding `platforms = expand_gfortran_versions(platforms)` to SuperLU_DIST's `build_tarballs.jl` script would solve the issue. I don't think this should be necessary, thought, but this might be a step towards debugging this.

I see that BinaryBuilder is using `MPItrampoline_jll v5.0.1+0`. There are newer versions available (5.2.1 is current). Maybe a newer version has this problem corrected? You would add a `compat=...` declaration for this, although I don't know how since the dependency on MPItrampoline is added implicitly via the call to `MPI.augment_platforms(platforms)`.

**Comment by JoshuaLampert** [comment]
We have the same issue for P4est in #6360. After adding `platforms = expand_gfortran_versions(platforms)` there is still an error for `mpitrampoline` with `libgfortran3`.

File: S/SuperLU_DIST/build_tarballs.jl [hunk 1]
```diff
@@ -93,6 +93,9 @@ platforms = filter(p -> !(p["mpi"] == "openmpi" && arch(p) == "armv6l" && libc(p
 platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && libc(p) == "musl"), platforms)
 platforms = filter(p -> !(p["mpi"] == "mpitrampoline" && Sys.isfreebsd(p)), platforms)
 
+platforms = expand_gfortran_versions(platforms)
```
Comment (giordano) [inline S/SuperLU_DIST/build_tarballs.jl:96]: Don't randomly expand libgfortran version if there's no need to.  This library doesn't even link to libgfortran

**Review by giordano — CHANGES_REQUESTED** [review]

File: S/SuperLU_DIST/build_tarballs.jl (hunk 1 above)
Comment (boriskaus) [reply S/SuperLU_DIST/build_tarballs.jl]: ok, any suggestions on how to link it versus a more recent version of `MPItrampoline_jll`?

Comment (giordano) [reply S/SuperLU_DIST/build_tarballs.jl]: No.  One (ideally not me, don't have much time) would have to investigate why dlopening fails.  #5028 is an example of process to track down these issues.

Comment (boriskaus) [reply S/SuperLU_DIST/build_tarballs.jl]: thanks; I'm afraid this is above my skillset..

**Comment by giordano** [comment]
For the record, the problem is that julia v1.7.3 comes with libgfortran4, but this build requires gcc 8, which is libgfortran5.  Solutions should be to either downgrade gcc to v7, or upgrade julia in CI to v1.8 (which has libgfortran5).  However now I believe again that https://github.com/JuliaPackaging/Yggdrasil/pull/5028#issuecomment-1166388492 might be a better way forward.

**Comment by giordano** [comment]
MPITrampoline is now passing audit.  However, the title of the PR is "update to version 1.8.2", but the PR is doing nothing.

**Comment by boriskaus** [comment]
Yes apologies; I initially wanted to bump the version in order to make it work with a newer version of PETSc, but realized meanwhile that the current PETSc version has issues on certain machines (like Apple silicon). Hence the rebuild first; new version will follow

**Pull request merged**